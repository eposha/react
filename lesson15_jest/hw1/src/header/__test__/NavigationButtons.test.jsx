import React from "react";
import { shallow } from "enzyme";
import NavigationButtons from "../NavigationButtons";

describe("<NavigationButtons/>", () => {
  it("Should display NavigationButtons", () => {
    const wrappedComponent = shallow(<NavigationButtons />);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it("Should call function by click on nav-button__next", () => {
    const mockOnClickNext = jest.fn();
    const wrappedComponent = shallow(
      <NavigationButtons onClick={mockOnClickNext} />
    );
    wrappedComponent.find(".nav-button__next").simulate("click");
    expect(mockOnClickNext).toBeCalled();
  });

  it("Should call function by click on nav-button__prev", () => {
    const mockOnClickPrev = jest.fn();
    const wrappedComponent = shallow(
      <NavigationButtons onClickPrev={mockOnClickPrev} />
    );
    wrappedComponent.find(".nav-button__prev").simulate("click");
    expect(mockOnClickPrev).toBeCalled();
  });
});
