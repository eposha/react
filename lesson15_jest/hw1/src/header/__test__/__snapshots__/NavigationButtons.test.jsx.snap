// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationButtons/> Should display NavigationButtons 1`] = `
<div
  className="nav-button"
>
  <button
    className="nav-button__prev button-style"
  >
    <i
      className="material-icons nav-button__size"
    >
      keyboard_arrow_left
    </i>
  </button>
  <button
    className="nav-button__next button-style"
  >
    <i
      className="material-icons nav-button__size "
    >
      chevron_right
    </i>
  </button>
</div>
`;
